blueocean-rest/src/main/java/io/jenkins/blueocean/rest/hal/Link.java

```diff
@@ -28,4 +29,27 @@ public String getHref(){
     public Link rel(String name) {
         return new Link(href+name);
     }
+
+
+    /**
+     * Gives ancestor Link, for example, for link("/a/b/c/d"), ancestor is "a/b/c/".
+     */
+    public Link ancestor(){
+        int i = href.lastIndexOf("/");
+
+        if(i>0) {
+            int j = href.substring(0, i).lastIndexOf("/");
```

Comment:
constructor has assertion on it. 

`assert this.href.endsWith("/");`

Adding the similar check might add to redundancy.